components/SendTransaction.tsx

```diff
@@ -35,14 +43,46 @@ export const SendTransaction = ({ setSuccess, success }: any) => {
           lamports: amount,
         })
       );
-      signature = await sendTransaction(transaction, connection);
-      await connection.confirmTransaction(signature, 'processed');
+
+      const {
+        context: { slot: minContextSlot },
+        value: { blockhash, lastValidBlockHeight },
+      } = await connection.getLatestBlockhashAndContext();
+
+      signature = await sendTransaction(transaction, connection, {
+        minContextSlot,
+      });
+
+      // ! deprecated method
+      // await connection.confirmTransaction(signature, 'processed');
+
+      await connection.confirmTransaction({
+        blockhash,
+        lastValidBlockHeight,
+        signature,
+      });
+
+      const nftId = await contract.nft.mintEmptyNft(connection, wallet);
+      console.log(
+        '🚀 ~ file: SendTransaction.tsx:62 ~ transfer ~ nftId',
+        nftId
+      );
+      await updateBackend(nftId, tiles);
       notify('Success', 'success');
       setSuccess(true);
     } catch (err: any) {
+      console.log(err);
       notify('Something went wrong. Please try again', 'error');
     }
-  }, [publicKey, sendTransaction, connection]);
+  }, [
```

Comment:
I think this is where the nftValueInSol was getting `undefined`. Even if there's value that was passed on the component. Since it's not part of the dependency list, the value will remain `undefined`.